main.tf

```diff
@@ -57,6 +71,14 @@ data "aws_iam_policy_document" "assume_role" {
   }
 }
 
+data "aws_subnet" "default" {
+  id = var.subnet_ids[0]
```

Comment:
this is to prevent introducing an additional required variable. happy to modify if that's a more preferred route.